web/struct_tree_layer_builder.js

```diff
@@ -74,21 +74,30 @@ const PDF_ROLE_TO_HTML_ROLE = {
 const HEADING_PATTERN = /^H(\d+)$/;
 
 class StructTreeLayerBuilder {
-  #treeDom = undefined;
+  #promise;
 
-  get renderingDone() {
-    return this.#treeDom !== undefined;
+  #treeDom;
+
+  #elementAttributes = new Map();
+
+  constructor(pdfPage) {
+    this.#promise = pdfPage.getStructTree();
   }
 
-  render(structTree) {
-    if (this.#treeDom !== undefined) {
+  async render() {
+    if (this.#treeDom) {
```

Comment:
This can lead to unnecessary re-parsing, please revert this.
Note how the `#walk`-method can return `null`, which the old condition thus handled correctly; see https://github.com/mozilla/pdf.js/blob/a41cd38387d09244c2a0c2405765efc8b3e9bce5/web/struct_tree_layer_builder.js#L120-L123